src/components/TodoItem/TodoItem.tsx

```diff
@@ -1,11 +1,23 @@
-import { memo } from "react";
+import React from "react";
+import { Action } from "redux";
 import { Item, Icons, Content } from "./TodoItem.styled";
 import TodoCheckbox from "../TodoCheckbox/TodoCheckbox";
-import PropTypes from "prop-types";
+import { TodoItemModel } from "../../App";
+import { Status } from "../../App";
 
-function TodoItem({ item, updateTodoItemStatus, onContentClick }) {
+const TodoItem: React.FC<{
+  item: TodoItemModel;
+  setTodoItemStatus: ({
+    itemId,
+    itemStatus,
+  }: {
+    itemId: string;
+    itemStatus: Status;
+  }) => Action;
```

Comment:
Should replace Action with void

component should not have any knowledge of redux